src/Workspaces/Core/Portable/CodeFixes/FixAllOccurrences/BatchFixAllProvider.cs

```diff
@@ -24,187 +22,222 @@ namespace Microsoft.CodeAnalysis.CodeFixes
     /// <summary>
     /// Helper class for "Fix all occurrences" code fix providers.
     /// </summary>
-    internal partial class BatchFixAllProvider : FixAllProvider
+    internal sealed class BatchFixAllProvider : AbstractDefaultFixAllProvider
     {
         public static readonly FixAllProvider Instance = new BatchFixAllProvider();
 
-        protected BatchFixAllProvider() { }
+        private BatchFixAllProvider()
+        {
+        }
 
-        #region "AbstractFixAllProvider methods"
+        protected sealed override string GetFixAllTitle(FixAllContext fixAllContext)
+            => FixAllContextHelper.GetDefaultFixAllTitle(fixAllContext);
 
-        public sealed override async Task<CodeAction?> GetFixAsync(FixAllContext fixAllContext)
+        protected sealed override async Task<Solution?> FixAllContextsAsync(
+            FixAllContext originalFixAllContext,
+            ImmutableArray<FixAllContext> fixAllContexts)
```

Comment:
this method is where BatchFixAllProvider and DocumentBasedFixAllProvider deviate.